src/stemmabench/algorithms/manuscript_in_tree_base.py

```diff
@@ -109,3 +105,16 @@ def build_text_lookup(self) -> Dict[str, "ManuscriptInTreeBase"]:
             for c in self.children:
                 out.update(c.build_text_lookup())
         return out
+    
+    def set_edges(self, edge_dict: Dict[str, float]) -> None:
+        """Sets the edges of the current manuscript and propagates the edge setting to all its children.
+        
+        ### Args:
+            - edge_dict (dict): The dictionary used to set the edges with the edges as keys and the edge distances as values.
+            The format of the keys is: "node_label1,node_label2".
+        """
+        if self.edges != []:
+            self._edges = []
+        for child in self.children:
+            self._edges.append(edge_dict[self.label + "," + child.label])
```

Comment:
f-string might be nicer here
```suggestion
            self._edges.append(edge_dict[f"{self.label},{child.label}"])
```

Reply:
F-string added